### PR TITLE
Better handle NPE from getting instance config

### PR DESCRIPTION
**Pull request by jackjlli**
Currently when `PinotHelixResourceManager` tries to get an instance config which doesn't exist, it will throw an NPE. This PR explicitly logs the instance Id for better debugging.

Current sample exception log (no instance information):
```
java.lang.NullPointerException
        at org.apache.helix.HelixProperty.<init>(HelixProperty.java:144)
        at org.apache.helix.model.InstanceConfig.<init>(InstanceConfig.java:76)
        at org.apache.pinot.controller.helix.core.PinotHelixResourceManager.getHelixInstanceConfig(PinotHelixResourceManager.java:289)
        at org.apache.pinot.controller.helix.core.PinotHelixResourceManager.getDataInstanceAdminEndpoints(PinotHelixResourceManager.java:2201)
        at org.apache.pinot.controller.util.TableSizeReader.getTableSubtypeSize(TableSizeReader.java:153)
```

New log:
```
Instance testInstance:1234_SERVER not found
```

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 1]
```diff
@@ -2203,13 +2202,18 @@ public boolean isSingleTenantCluster() {
    * http requests. So, on response, we need mapping from the endpoint to the
    * server instances. With BiMap, both mappings are easily available
    */
-  @Nonnull
-  public BiMap<String, String> getDataInstanceAdminEndpoints(@Nonnull Set<String> instances)
-      throws InvalidConfigException {
+  public @Nonnull
+  BiMap<String, String> getDataInstanceAdminEndpoints(@Nonnull Set<String> instances)
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Please revert the changes for the top 2 lines

File: pinot-common/src/main/java/org/apache/pinot/common/exception/InstanceNotFoundException.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.exception;
+
+public class InstanceNotFoundException extends Exception {
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/exception/InstanceNotFoundException.java]: I don't think we need this new exception type

File: pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/TableSizeReaderTest.java [hunk 3]
```diff
@@ -224,21 +223,21 @@ public void run() {
   }
 
   @Test
-  public void testNoSuchTable()
-      throws InvalidConfigException {
-    TableSizeReader reader = new TableSizeReader(executor, connectionManager, _controllerMetrics, helix);
+  public void testNoSuchTable() throws Exception {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/TableSizeReaderTest.java]: Please reformat these files using the Pinot Style

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 4]
```diff
@@ -2203,13 +2202,18 @@ public boolean isSingleTenantCluster() {
...
+      throws InstanceNotFoundException, InvalidConfigException {
     Preconditions.checkNotNull(instances);
     BiMap<String, String> endpointToInstance = HashBiMap.create(instances.size());
     for (String instance : instances) {
       InstanceConfig helixInstanceConfig = getHelixInstanceConfig(instance);
+      if (helixInstanceConfig == null) {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Maybe log an warning and skip this instance?

Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Done

File: pinot-common/src/main/java/org/apache/pinot/common/exception/InstanceNotFoundException.java (hunk 2 above)
Comment (jackjlli) [reply pinot-common/src/main/java/org/apache/pinot/common/exception/InstanceNotFoundException.java]: Removed.

File: pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/TableSizeReaderTest.java (hunk 3 above)
Comment (jackjlli) [reply pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/TableSizeReaderTest.java]: Done.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 5]
```diff
@@ -284,8 +285,13 @@ public HelixAdmin getHelixAdmin() {
    * @return Helix instance config
    */
   @Nonnull
-  public InstanceConfig getHelixInstanceConfig(@Nonnull String instanceId) {
+  public InstanceConfig getHelixInstanceConfig(@Nonnull String instanceId) throws InstanceNotFoundException {
     ZNRecord znRecord = _cacheInstanceConfigsDataAccessor.get("/" + instanceId, null, AccessOption.PERSISTENT);
+    if (znRecord == null) {
```
[the remark this answers is not in the document]
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Done.
